### PR TITLE
Fixed metadata validation for NonSupported and Legacy packs

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28689

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -630,6 +631,17 @@ def validate_added_files(self, added_files, modified_files):
...
+    @staticmethod
+    def should_raise_pack_version(pack):
+        """
+        Args:
+            pack: The pack name.
+
+        Returns: False if pack is in IGNORED_PACK_NAMES or ApiModules else True.
+
+        """
+        return not (pack in IGNORED_PACK_NAMES or 'ApiModules' in pack)
```
Comment (Itay4) [inline demisto_sdk/commands/validate/validate_manager.py:643]: did you consider adding `ApiModules` to `IGNORED_PACK_NAMES` ?

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -630,6 +631,17 @@ def validate_added_files(self, added_files, modified_files):
                                                          added_files=added_files))
         return all(valid_files)
 
+    @staticmethod
+    def should_raise_pack_version(pack):
```
Comment (Itay4) [inline demisto_sdk/commands/validate/validate_manager.py:635]:
```suggestion
    def should_raise_pack_version(pack: str) -> book:
```

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 3]
```diff
@@ -1053,3 +1053,21 @@ def test_add_origin(branch_name, prev_ver, expected):
...
+
+
+@pytest.mark.parametrize('pack_name, expected', [
+    ('NonSupported', False),
+    ('PackName1', True)
+])
+def test_should_raise_pack_version(pack_name, expected):
+    """
+    Given
+        - Pack name.
```
Comment (Itay4) [inline demisto_sdk/commands/validate/tests/validators_test.py:1065]: please add here the cases

**Comment by Itay4** [comment]
please update changelog

File: demisto_sdk/commands/validate/validate_manager.py [hunk 4]
```diff
@@ -630,6 +631,17 @@ def validate_added_files(self, added_files, modified_files):
...
+    @staticmethod
+    def should_raise_pack_version(pack: str) -> bool:
+        """
+        Args:
+            pack: The pack name.
+
+        Returns: False if pack is in IGNORED_PACK_NAMES else True.
+
+        """
+        return pack not in IGNORED_PACK_NAMES
```
Comment (Itay4) [inline demisto_sdk/commands/validate/validate_manager.py:643]: what about `ApiModules` ?

Comment (Shellyber) [reply demisto_sdk/commands/validate/validate_manager.py:643]: I added it to IGNORED_PACK_NAMES as requested

Comment (Itay4) [reply demisto_sdk/commands/validate/validate_manager.py:643]: i think you forgot to push it then

Comment (Shellyber) [reply demisto_sdk/commands/validate/validate_manager.py:643]: yeah, just noticed. Pushed it now

File: demisto_sdk/commands/common/constants.py [hunk 5]
```diff
@@ -728,7 +728,8 @@ class FileType(Enum):
 TESTS_DIRECTORIES = [
     'testdata',
     'test_data',
-    'data_test'
+    'data_test',
+    'tests_data'
```
Comment (Itay4) [inline demisto_sdk/commands/common/constants.py:732]: is this change intended ?

**Comment by Itay4** [comment]
@Shellyber please resolve conflicts in the changelog

File: demisto_sdk/commands/common/constants.py (hunk 5 above)
Comment (Shellyber) [reply demisto_sdk/commands/common/constants.py:732]: Not to this pr but basically yes.
It to not run validation on this folder in content

**Review by Itay4 — APPROVED** [review]
